### PR TITLE
(INT): Match to if let

### PR DESCRIPTION
**Pull request by ujpv**
New intention.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/intentions/MatchToIfLetIntention.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if (notVoidArms.size == 1 &&
+            notVoidArms[0].matchPat.patList.size == 1 &&
+            notVoidArms[0].matchPat.patList[0].text != "_") {
+            return matchExpr
+        } else {
+            return null
+        }
+    }
+
+    fun RustExprElement.notVoid(): Boolean =
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/intentions/MatchToIfLetIntention.kt]: I think we want to treat `{}` and `()` the same. I'd write this as

```
val RustExprElement.isVoid: Boolean get() =
   (this is RustBlockExpeElement && (block?.lbrace.getNextNonCOmmentSIbling()? == block.rbrace)) ||
    this is RustUnitExpr
```

Patterns like

```
if (cond) false else true
```

can be simplified to just `!cond`.

File: src/test/kotlin/org/rust/ide/intentions/MatchToIfLetIntentionTest.kt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    fun testUnavailable2() = doUnavailableTest(
+        """
+        enum MyOption {
+            Nothing,
+            Some(x),
+        }
+
+        fn main() {/*caret*/
+            let a = MyOption::Some(52);
```
Comment (matklad) [inline src/test/kotlin/org/rust/ide/intentions/MatchToIfLetIntentionTest.kt]: I think the caret should be inside a match expr to really test the case of non-void arms.

File: src/main/kotlin/org/rust/ide/intentions/MatchToIfLetIntention.kt (hunk 1 above)
Comment (ujpv) [inline src/main/kotlin/org/rust/ide/intentions/MatchToIfLetIntention.kt]: Thanks!
I have simplified it.

**Comment by matklad** [comment]
LGTM!

**Pull request merged**